src/Compiler/WebFormsCompilerExtensions.cs

```diff
@@ -0,0 +1,45 @@
+// MIT License.
+
+using System.Web.UI;
+using Microsoft.AspNetCore.SystemWebAdapters;
+using Microsoft.Extensions.Options;
+using WebForms.Compiler;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public static class WebFormsCompilerExtensions
+{
+    public static void AddWebFormsCompilation(this IServiceCollection services, Action<PageCompilationOptions> configure)
+    {
+        services.AddWebFormsCompilationCore(configure);
+        services.AddSingleton<PersistentSystemWebCompilation>();
+        services.AddSingleton<IPageCompiler>(ctx => ctx.GetRequiredService<PersistentSystemWebCompilation>());
+        services.AddSingleton<IWebFormsCompiler>(ctx => ctx.GetRequiredService<PersistentSystemWebCompilation>());
+    }
+
+    public static ISystemWebAdapterBuilder AddDynamicWebForms(this ISystemWebAdapterBuilder services, Action<PageCompilationOptions> configure)
+    {
+        services.Services.AddWebFormsCompilationCore(configure);
+        services.Services.AddHostedService<WebFormsCompilationService>();
+        services.Services.AddSingleton<IPageCompiler, DynamicSystemWebCompilation>();
+
+        return services;
+    }
+
+    private static void AddWebFormsCompilationCore(this IServiceCollection services, Action<PageCompilationOptions> configure)
+    {
+        services.AddOptions<PageCompilationOptions>()
+            .Configure(configure);
+
+        services.AddOptions<PagesSection>()
+            .Configure<IOptions<PageCompilationOptions>>((options, compilation) =>
+            {
+                foreach (var known in compilation.Value.KnownTags)
+                {
+                    options.DefaultTagNamespaceRegisterEntries.Add(known);
+                }
+
+                options.EnableSessionState = System.Web.Configuration.PagesEnableSessionState.True;
```

Comment:
it's ok now.. but I am thinking if at page level someone has disabled, how that will work ? (If you think we need to handle that, please create an issue)

Reply:
I believe it should already be handled. This is the initial option to add it; if we see issues there we can raise issues

Reply:
If it's disabled on a page, I belive that should be working